gatorgrade/input/set_up_shell.py

```diff
@@ -0,0 +1,28 @@
+"""Set-up the shell commands."""
+import os, sys
+
+
+def run_setup(front_matter):
+    """Run the shell set up commands and exit the program if a command fails.
+
+    Args:
+        front_matter: A dictionary whose 'setup' key contains the set up commands
+        as a multi-line string.
+
+    """
+    print("Running set up commands...")
+    for line in front_matter["setup"].splitlines():
```

Comment:
`setup` is not a required attribute in the first document in a `gatorgrade.yml` file, so it may not exist. You should probably handle that case here (by just not doing anything if it is not defined).